plugins/inputs/dmcache/dmcache_linux.go

```diff
@@ -0,0 +1,131 @@
+// +build linux
+
+package dmcache
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+)
+
+const metricName = "dmcache"
+
+var fieldNames = [...]string{
+	"metadata_used",
+	"metadata_free",
+	"cache_used",
+	"cache_free",
+	"read_hits",
+	"read_misses",
+	"write_hits",
+	"write_misses",
+	"demotions",
+	"promotions",
+	"dirty",
+}
+
+func (c *DMCache) Gather(acc telegraf.Accumulator) error {
+	outputLines, err := c.rawStatus()
+	if err != nil {
+		return err
+	}
+
+	var total map[string]interface{}
+	if !c.PerDevice {
+		total = make(map[string]interface{})
+	}
+
+	for _, s := range outputLines {
+		fields := make(map[string]interface{})
+		data, err := parseDMSetupStatus(s)
+		if err != nil {
+			return err
+		}
+
+		for _, f := range fieldNames {
+			fields[f] = calculateSize(data, f)
+		}
+
+		if c.PerDevice {
+			tags := map[string]string{"device": data["device"].(string)}
+			acc.AddFields(metricName, fields, tags)
+		} else {
+			aggregateStats(total, fields)
+		}
+	}
+
+	if !c.PerDevice {
+		acc.AddFields(metricName, total, nil)
+	}
+
+	return nil
+}
+
+func parseDMSetupStatus(line string) (status map[string]interface{}, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			status = nil
+			err = r.(error)
+		}
+	}()
+
+	values := strings.Split(line, " ")
+	status = make(map[string]interface{})
+
+	status["device"] = values[0][:len(values[0])-1]
```

Comment:
or you can check that the array is of a minimum length, and just ignore all metrics if it isn't.